addons/product/models/product_product.py

```diff
@@ -392,7 +392,7 @@ def unlink(self):
             if product.image_variant_1920 and not product.product_tmpl_id.image_1920:
                 product.product_tmpl_id.image_1920 = product.image_variant_1920
             # Check if the product is last product of this template...
-            has_other_products = product_ids_by_template_id[product.product_tmpl_id.id] - {product.id}
+            has_other_products = product_ids_by_template_id and product_ids_by_template_id[product.product_tmpl_id.id] - {product.id}
```

Comment:
Try to delete two products, one active and one archived, you will still have the problem
You may rather use a mix between `active_test=False` and `.get` on the dict

Also, can you add a test (with a deletion of several products) ?